src/Effect.cpp

```diff
@@ -298,28 +298,25 @@ Effect::add_external_effect(const Effect &e, std::vector<const Block*> call_chai
 }
 
 /*
- * TODO: tried doing this with `find', but I couldn't get all the `const's
- * right.  Gave up; wrote it by hand!
+ * 
+ *
  */
 bool
 Effect::is_read(const Variable *v) const
 {
-	vector<Variable *>::size_type len = read_vars.size();
-	vector<Variable *>::size_type i;
-
-	for (i = 0; i < len; ++i) {
-		if (read_vars[i] == v) {
-			return true;
+	if (std::find(read_vars.begin(), read_vars.end(), v) != read_vars.end()) {
+		return true;
+	}
+	else {
```

Comment:
I think Yang meant we don't need the "else {" and "}" enclosing because by
the time the execution reaches the body of the else branch, the
if-condition must be false.

It's a style issue. Some people prefer to make the else part explicit to be
more readable. I am not against that idea.

BTW, I think there are a few more places in Csmith using iterations to find
certain items in vectors or hash tables. Can you replace all of them with
std::find or map::find?

-Xuejun



On Mon, Oct 26, 2015 at 10:07 PM, Yang Chen <notifications@github.com>
wrote:

> In src/Effect.cpp
> <https://github.com/csmith-project/csmith/pull/21#discussion_r43084784>:
>
> >   */
> >  bool
> >  Effect::is_read(const Variable *v) const
> >  {
> > -	vector<Variable *>::size_type len = read_vars.size();
> > -	vector<Variable *>::size_type i;
> > -
> > -	for (i = 0; i < len; ++i) {
> > -		if (read_vars[i] == v) {
> > -			return true;
> > +	if (std::find(read_vars.begin(), read_vars.end(), v) != read_vars.end()) {
> > +		return true;
> > +	}
> > +	else {
>
> Seems we don't need the else branch, and hence we don't need to change
> the code below?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/csmith-project/csmith/pull/21/files#r43084784>.
>